packages/astro/src/core/app/index.ts

```diff
@@ -137,6 +146,42 @@ export class App {
 		});
 	}
 
+	/**
+	 * Content types that can be passed when sending a request via a form
+	 *
+	 * https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/enctype
+	 * @private
+	 */
+	#formContentTypes = ['application/x-www-form-urlencoded', 'multipart/form-data', 'text/plain'];
+
+	/**
+	 * Returns a middleware function in charge to check the `origin` header.
+	 *
+	 * @private
+	 */
+	#createOriginCheckMiddleware(): MiddlewareHandler {
+		return defineMiddleware((context, next) => {
+			const { request, url } = context;
+			const contentType = request.headers.get('content-type');
+			if (contentType) {
+				if (this.#formContentTypes.includes(contentType)) {
```

Comment:
I think the `content-type` here may not be thoroughly checked. There was a [SvelteKit vulnerability](https://github.com/sveltejs/kit/security/advisories/GHSA-gv7g-x59x-wf8f) where `content-type` value could have different casing, which can bypass the check.

Reply:
> Can this be fixed by doing `contentType.toLowerCase()`?